spec/lang/representation.md

```diff
@@ -92,8 +93,17 @@ This is required to achieve a "monotonicity" with respect to provenance (as disc
 
 ### Pointers
 
+Pointers are significantly more complex to represent than just the integer address.
+For one, they need to encode the provenance and decode it only when
+all bytes match.
+On the other hand, some pointers are wide pointers which also need
+to encode their metadata.
+The helpers `decode_ptr` and `encode_ptr` deal with the thin pointers,
+and the `decode_ptr_meta` helper inferrs the metadata kind from the
+pointee layout.
```

Comment:
Elsewhere in this file we are using one-(sub)sentence-per-line, please stick to that style.